ui/dashboards/src/templating/components.tsx

```diff
@@ -0,0 +1,164 @@
+// Copyright 2022 The Perses Authors
```

Comment:
Nit: would be great to break these out into the appropriate folders once you think you have the logic down (e.g. `components`, `model`, etc.) Same with the `core` file and other stuff in this `templating` folder.

Reply:
Ack - planning on doing that in the non-draft PR. I was trying to limit the merge conflicts while prototyping. That being said, at some point do you think it will make sense to take a [feature specific file approach](https://reactjs.org/docs/faq-structure.html#grouping-by-features-or-routes) as we grow so files are closer to the modules that need them? I've always struggled with this lol 🤷